src/main/java/homework_4/custom_file_reader/Main.java

```diff
@@ -0,0 +1,9 @@
+package homework_4.custom_file_reader;
+
+public class Main {
+
+    public static void main(String[] args) {
+        new CustomFileReader().run1();
```

Comment:
2nd case somehow broken: 
![image](https://user-images.githubusercontent.com/58872509/134049209-b2a2a4f1-802c-4a1e-a732-124b00bb00fb.png)

Reply:
Nice usage of abstractions!